### DISCUSSION TITLE
Changing names for RGB wallet and standard library

### DISCUSSION THREAD

**Discussion by dr-orlovsky** [RGB-WG/rgb#58] [General]
Right now we have two RGB repositories providing client-facing APIs: `rgb-wallet` and `rgb`. However, this naming is confusing in multiple ways.

`rgb-wallet` repository provides
- `rgb-std` crate (having no "wallet" functionality and not depending on `rust-bitcoin`)
- `rgb-invoice` crate (since v0.11 will also not depend on `rust-bitcoin`)
- `rgb-psbt` crate (since v0.11 will also not depend on `rust-bitcoin` and supporting PSBTv2)

The last two crates were the reason for the naming since in `v0.10` since they were constituting a single `rgb-wallet` crate inside the repo. However, with `v0.11` I am splitting them since many projects will need just invoice parsing/generator functionality without requiring the whole RGB consensus + standard library + PSBT functionality. Also, some wallet projects may need invoicing and the standard library, and will use some alternative PSBT like `rust-bitcoin` + BDK + rgb tools, etc.

In other words, there is no "wallet" in `rgb-wallet` repo: we can't filter RGB data from the standard library for a specific wallet descriptor or generate a PSBT spending some existing RGB assignments.

Now, on the other hand, the other repo `rgb` (the most user/client-facing repo) has received the actual wallet facilities allowing all of that functionality and providing a command-line wallet tool able to work with descriptors and do the filtering + generate PSBTs. Still this command-line tool has a crate name `rgb-contracts` (since just `rgb` is not available and is used by some color-managing lib), which is confusing (since it is not "contracts" but an actual command-line RGB wallet).

So I propose to rename `rgb-wallet` repository into `rgb-std` - and use `rgb-wallet` as a name for the crate containing the command-line wallet tool - so it can be installed by users with just `cargo install rgb-wallet` command (instead of `cargo install rgb-contracts --all-features` used today).

This will also reflect the same approach I used for @BP-WG in v0.11 stack: `bp-std` repo and `bp-wallet` repo, with the last one providing the actual wallet APIs (runtime) and a command-line tool wrapping it.

**Comment by fedsten** [comment]
Personally I agree that the current situation is a bit confusing and a clean up would have benefits, but at the same time I am worried about using the same name `rgb-wallet` for a repository that is unrelated to what is currently named `rgb-wallet` as it can create a lot of confusion to developers currently working on RGB, and unfortunately we can no longer assume that they are all aware of these discussions. Do you think we can find another name that is not currently used in any other repository?

**Reply by dr-orlovsky** [reply]
With a semantic versioning the rename will not affect them unless they manually bump the crate version. Then, a compiler-generated deprecation warning can work to notify them about a proper way to upgrade (plus READMEs)
